src/kernel/emulation/linux/misc/sysctl_machdep.c

```diff
@@ -51,3 +89,131 @@ sysctl_handler(handle_brand_string)
 	return 0;
 }
 
+sysctl_handler(handle_max_basic)
+{
+	setup(0);
+
+
+    __cpuid(level, eax, ebx, ecx, edx);
+
+
+
+
+	copyout_int(eax, (char*)old,oldlen);
+
+	return 0;
+}
+
+sysctl_handler(handle_family)
+{
+    setup(1);
+
+    __cpuid(level, eax, ebx, ecx, edx);
+
+    eax = eax >> 7;
+    eax &= 15;
+
+    copyout_string(old,"%d",eax);
+
+    return 0;
+}
+
+sysctl_handler(handle_model)
+{
+    setup(1);
+
+    __cpuid(level, eax, ebx, ecx, edx);
+
+    eax = eax >> 3;
+    eax &= 15;
+
+    copyout_int(eax,(char*)old, oldlen);
+
+    return 0;
+}
+
+sysctl_handler(handle_stepping)
+{
+    setup(1);
+
+    __cpuid(level, eax, ebx, ecx, edx);
+
+    eax &= 15;
+
+    copyout_int(eax,(char*)old, oldlen);
+
+    return 0;
+}
+
+sysctl_handler(handle_brand_string)
+{
+    setup(0x80000000);
+
+    __cpuid(level,eax,eabx, eacx, eadx);
+
+    if(eax < 0x80000004) // the information is not implemented
+        return 2;
+
+
+
+    union
+    {
+        unsigned int brand[12];
+        char name[49];
+    } v;
+
+		for (int i = 1; i < 4; i++)
+		{
+				eax = 0x80000000+i;
+				__cpuid(level, eax, ebx, ecx, edx);
+				v.brand[0x0+(i-1)*4] = eax;
+				v.brand[0x1+(i-1)*4] = ebx;
+				v.brand[0x2+(i-1)*4] = ecx;
+				v.brand[0x3+(i-1)*4] = edx;
+		}
+
+    v.name[48] = 0;
+    copyout_string(v.name, (char*) old, oldlen);
+
+    return 0;
+}
+sysctl_handler(handle_features)
+{
+
+    if(old != NULL)
+    {
+        setup(1);
+
+        char features[][] = {"FPU","VME", "DE", "PSE", "TSC", "MSR", "PAE", "MCE", "CX8", "APIC", "","SEP","MTRR","PGE",
```

Comment:
Can't have an array of arrays of undefined length; use:
```suggestion
        cosnt char* features[] = {"FPU","VME", "DE", "PSE", "TSC", "MSR", "PAE", "MCE", "CX8", "APIC", "","SEP","MTRR","PGE",
```

Reply:
You actually can, like this:
`char features[][5]`
or whatever the max size is. But I'm not sure that's better than `const char* features[]`

Reply:
Well yeah, I meant `char[][]` isn't valid because the values inside the parent array have to have the same type and length. That's perfectly fine (in this case, it would be `char[][7]` to include the null terminator).

But we don't need to copy constant strings into an array since we're not going modify them anyways.

Reply:
`static const char features[][7]` will actually generate one contiguous character buffer and will not do any copying.